src/Components/Web/src/Forms/InputDate.cs

```diff
@@ -30,12 +35,31 @@ public class InputDate<TValue> : InputBase<TValue>
         /// </summary>
         [DisallowNull] public ElementReference? Element { get; protected set; }
 
+        /// <inheritdoc />
+        public override Task SetParametersAsync(ParameterView parameters)
+        {
+            _typeAttributeValue = parameters.TryGetValue<string>("type", out var typeAttributeValue)
+                ? typeAttributeValue
+                : "date";
+
+            _format = _typeAttributeValue switch
+            {
+                "date" => DateFormat,
+                "datetime-local" => DateTimeLocalFormat,
+                "month" => MonthFormat,
+                "time" => TimeFormat,
+                _ => throw new InvalidOperationException($"Unsupported 'type' attribute value '{_typeAttributeValue}'.")
+            };
+
+            return base.SetParametersAsync(parameters);
```

Comment:
Rather than run `parameters.TryGetValue` to find `"type"` above, would it work to call `base.SetParametersAsync` first, and then read the `"type"` entry out of `AdditionalAttributes`?

Reply:
Also, more broadly, should we consider *not* using the `type` attribute for this, but instead having a parameter like:

```cs
[Parameter] public InputDateType Type { get; set; } = InputDateType.Date;

enum InputDateType { Date, Time, DateTime, Month }
```

I'm asking because it's unclear that there's an advantage to coupling this to the HTML `type` strings. They won't appear in intellisense, and TBH their naming is nonobvious and hard to remember (specifically, `datetime-local` - local to the server or the client? It's not clear for something like Blazor Server).

Since it's never been possible to pass a regular HTML `type` attribute to this component, because we always overrode it, it wouldn't be breaking if we continued to ignore custom `type` attributes and used a separate .NET parameter instead.

Reply:
I'm not 100% set either way on this and can see arguments in both directions. If you have a clear preference @MackinnonBuck then let's go with that since you've been thinking about it the most.

Reply:
How do we feel about having a `[Parameter, EditorRequired] public string Type { get; set; }` where the `string` is one of the regular values? I'm somewhat ambivalent about adding a new enum type for this, but we should flag that specifying `Type` is required.

Reply:
Why would specifying `Type` be required? For back-compat, we should default to `date`.

Reply:
Using an enum parameter like `InputDateType` was something I considered, but my primary reasoning for looking at the `type` attribute instead was mainly to keep the API surface small and to reuse existing web concepts developers might already be familiar with. However, it also makes the feature difficult to discover, and it breaks the pattern that we've been following of abstracting away the `type` attribute. So, I think having an `InputDateType` might actually be the better way to go.